test/scalecube-cluster/cluster.test.js

```diff
@@ -0,0 +1,134 @@
+import { LogicalCluster as Cluster } from 'src/scalecube-cluster/LogicalCluster/LogicalCluster';
+import 'rxjs/add/operator/zip';
+import 'rxjs/add/operator/do';
+import 'rxjs/add/operator/elementAt';
+
+const expectMessage = (cluster, at, messageExpected) =>
+    cluster
+        .listenMembership()
+        .elementAt(at)
+        .do(messageSent=>expect(messageSent).toEqual(messageExpected))
+        .subscribe();
+
+// TODO messages order isn't important
+// tests can be changed to have message in different order
+// this was easier to implemented
+
+describe('Cluster suite', () => {
+  const createClusters = () => {
+
+    const clusterA = new Cluster();
+    const clusterB = new Cluster();
+    const clusterC = new Cluster();
+
+    clusterA.metadata('clusterA');
+    clusterB.metadata('clusterB');
+    clusterC.metadata('clusterC');
+
+    return {clusterA, clusterB, clusterC};
+  };
+
+  it('When clusterA join clusterB and B join C all cluster should be on all clusters', () => {
+    const {clusterA, clusterB, clusterC} = createClusters();
+
+    clusterA.join(clusterB);
+    clusterB.join(clusterC);
+
+    expect(clusterA.members().map(i=>i.metadata())).toEqual([ 'clusterA', 'clusterB', 'clusterC' ]);
+    expect(clusterB.members().map(i=>i.metadata())).toEqual([ 'clusterB', 'clusterA', 'clusterC' ]);
+    expect(clusterC.members().map(i=>i.metadata())).toEqual([ 'clusterC', 'clusterB', 'clusterA' ]);
+  });
+  it('When clusterA shutdown clusterB and C should not have clusterA and remove message should be send', () => {
+    const {clusterA, clusterB, clusterC} = createClusters();
+    expect.assertions(8);
+
+
+    clusterA.join(clusterB);
+    clusterB.join(clusterC);
+
+      [clusterA,clusterB,clusterC].forEach(cluster =>
+          cluster
+              .listenMembership()
+              .do((msg) => {
+                  expect(msg.metadata).toEqual({});
+                  expect(msg.type).toBe('remove');
+              })
+              .subscribe()
+      );
+
+    clusterA.shutdown();
```

Comment:
Is it OK that after clusterA.shutdown() method is called the clusterA removed from members but still exists with its 'myId' and 'metadata'?

Reply:
I tired to delete the instance from it's self. But it's not really working with JS.
Any ideas? 
Maybe we need to tweak the behavior

Reply:
@idanilt basically for now it's not critical. I'll have a look and think about ways to handle it but I think that PR can be merged 🙂